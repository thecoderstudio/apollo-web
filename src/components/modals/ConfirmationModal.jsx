import React from 'react';
import Modal from './modal';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import OutlinedButton from '../buttons/OutlinedButton';
import Button from '../buttons/Button';
import { browserPreferredTheme } from '../../theme';

const propTypes = {
  title: PropTypes.string.isRequired,
  confirmationCallback: PropTypes.func.isRequired,
  cancelCallback: PropTypes.func.isRequired,
  confirmationButtonColor: PropTypes.string,
  confirmationButtonText: PropTypes.string
};

const defaultProps = {
  confimationButtonColor: browserPreferredTheme.primary,
  confirmationButtonText: 'confirm'
}

const Content = styled.div`
  display: grid;
  grid-template-columns: [cancel] 1fr [delete] 1fr;
`;

const CancelButton = styled(OutlinedButton)`
  margin: 20px;
  grid-column: cancel;
`;

const ConfirmButton = styled(Button)`
  grid-column: delete;
  margin: 20px;
  background-color: ${props => props.color}
`;

export default function ConfirmationModal(props) {
  return(
    <Modal visible={true} title={props.title}>
      <Content>
        <CancelButton onClick={props.cancelCallback}>cancel</CancelButton>
        <ConfirmButton color={props.confirmationButtonColor} onClick={props.confirmationCallback}>{props.confirmationButtonText}</ConfirmButton>
      </Content>
    </Modal>
  );
}

ConfirmationModal.propTypes = propTypes;
ConfirmationModal.defaultProps = defaultProps;