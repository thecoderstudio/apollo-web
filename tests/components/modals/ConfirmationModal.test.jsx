import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import ConfirmationModal from '../../../src/components/modals/ConfirmationModal';

function getComponent(props) {
  return renderer.create(
    <Provider store={configureStore([])({})}>
      <ConfirmationModal {...props}  />
    </Provider>
  );
}

describe('ConfirmationModal', () => {
  const confirmSpy = jest.fn();
  const cancelSpy = jest.fn();
  const props = {
    title: 'title',
    confirmationCallback: confirmSpy,
    closeModalFunction: cancelSpy,
  };

  it('renders correctly', () => {
    expect(getComponent(props).toJSON()).toMatchSnapshot();
  });

  it('calls delete function', () => {
    const component = getComponent(props);
    component.root.findAllByType('button')[0].props.onClick();
    expect(cancelSpy).toHaveBeenCalled();
  });

  it('calls cancel button', () => {
    const component = getComponent(props);
    component.root.findAllByType('button')[1].props.onClick();
    expect(confirmSpy).toHaveBeenCalled();
  });
});