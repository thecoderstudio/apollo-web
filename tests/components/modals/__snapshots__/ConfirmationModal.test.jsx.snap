// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal renders correctly 1`] = `
.c0 {
  display: inline;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
}

.c4 {
  display: grid;
  grid-template-columns: [cancel] 1fr [confirm] 1fr;
}

.c5 {
  max-height: 50px;
  padding: 12px;
  border: none;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid;
  margin: 20px;
  grid-column: cancel;
}

.c5:hover {
  opacity: 0.90;
}

.c5:focus {
  outline: none;
}

.c5:active {
  opacity: 0.80;
}

.c6 {
  max-height: 50px;
  padding: 12px;
  border: none;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
  grid-column: confirm;
  margin: 20px;
}

.c6:hover {
  opacity: 0.90;
}

.c6:focus {
  outline: none;
}

.c6:active {
  opacity: 0.80;
}

.c1 {
  box-shadow: 0 10px 15px -3px rgba(0,0,0,0.1),0 4px 6px -2px rgba(0,0,0,0.05);
  padding: 40px;
  border-radius: 10px;
  display: grid;
  grid-template-rows: [title] 1fr [content] 1.5fr;
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 100%;
  max-width: 400px;
}

.c2 {
  grid-row: title;
  width: 75%;
  margin: 0 auto;
  text-align: center;
}

.c3 {
  grid-row: content;
  padding-top: 20px;
}

<div
  className="c0"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <h2
      className="c2"
    >
      title
    </h2>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <button
          className="c5"
          onClick={[MockFunction]}
        >
          Cancel
        </button>
        <button
          className="c6"
          onClick={[MockFunction]}
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
